File: shop-frontend/src/styles/theme.js
// Базовая тема по умолчанию (fallback)
export const defaultTheme = {
    colors: {
        primary: '#3498db',
        secondary: '#2ecc71',
        accent: '#e74c3c',
        text: '#333333',
        background: '#ffffff',
        buttonText: '#ffffff',
        footer: {
            background: '#2c3e50',
            text: '#ecf0f1'
        },
        status: {
            success: '#27ae60',
            error: '#c0392b',
            warning: '#f39c12',
            info: '#3498db'
        },
        interactive: {
            hover: '#34495e',
            active: '#2980b9'
        }
    },
    typography: {
        fontFamily: 'Roboto, sans-serif',
        headingFontFamily: 'Montserrat, sans-serif',
        bodyFontFamily: 'Open Sans, sans-serif',
        fontSize: {
            tiny: '0.75rem',
            small: '0.875rem',
            base: '1rem',
            large: '1.25rem',
            heading: '1.5rem',
            display: '2rem'
        },
        fontWeight: {
            light: 300,
            normal: 400,
            medium: 500,
            bold: 700
        }
    },
    spacing: {
        xs: '0.25rem',
        sm: '0.5rem',
        md: '1rem',
        lg: '1.5rem',
        xl: '2rem',
        xxl: '3rem'
    },
    borderRadius: {
        small: '2px',
        button: '4px',
        card: '8px',
        input: '4px',
        image: '8px'
    },
    shadows: {
        small: '0 1px 2px rgba(0,0,0,0.1)',
        card: '0 2px 4px rgba(0,0,0,0.1)',
        medium: '0 4px 6px rgba(0,0,0,0.1)',
        large: '0 4px 8px rgba(0,0,0,0.1)'
    },
    assets: {
        logoUrl: null,
        headerImageUrl: null,
        footerLogoUrl: null,
        faviconUrl: null
    }
};

// Функция для создания темы из API ответа
export const createThemeFromApi = (apiTheme) => {
    if (!apiTheme) return defaultTheme;

    return {
        colors: {
            primary: apiTheme.primaryColor || defaultTheme.colors.primary,
            secondary: apiTheme.secondaryColor || defaultTheme.colors.secondary,
            accent: apiTheme.accentColor || defaultTheme.colors.accent,
            text: apiTheme.textColor || defaultTheme.colors.text,
            background: apiTheme.backgroundColor || defaultTheme.colors.background,
            buttonText: apiTheme.buttonTextColor || defaultTheme.colors.buttonText,
            footer: {
                background: apiTheme.footerBackgroundColor || defaultTheme.colors.footer.background,
                text: apiTheme.footerTextColor || defaultTheme.colors.footer.text
            },
            status: {
                success: apiTheme.successColor || defaultTheme.colors.status.success,
                error: apiTheme.errorColor || defaultTheme.colors.status.error,
                warning: apiTheme.warningColor || defaultTheme.colors.status.warning,
                info: apiTheme.infoColor || defaultTheme.colors.status.info
            },
            interactive: {
                hover: apiTheme.hoverColor || defaultTheme.colors.interactive.hover,
                active: apiTheme.activeColor || defaultTheme.colors.interactive.active
            }
        },
        typography: {
            fontFamily: apiTheme.fontFamily || defaultTheme.typography.fontFamily,
            headingFontFamily: apiTheme.headingFontFamily || defaultTheme.typography.headingFontFamily,
            bodyFontFamily: apiTheme.bodyFontFamily || defaultTheme.typography.bodyFontFamily,
            fontSize: defaultTheme.typography.fontSize,
            fontWeight: defaultTheme.typography.fontWeight
        },
        spacing: defaultTheme.spacing,
        borderRadius: {
            small: defaultTheme.borderRadius.small,
            button: apiTheme.buttonRadius || defaultTheme.borderRadius.button,
            card: apiTheme.cardRadius || defaultTheme.borderRadius.card,
            input: apiTheme.inputRadius || defaultTheme.borderRadius.input,
            image: defaultTheme.borderRadius.image
        },
        shadows: {
            small: defaultTheme.shadows.small,
            card: apiTheme.cardShadow || defaultTheme.shadows.card,
            medium: apiTheme.boxShadow || defaultTheme.shadows.medium,
            large: defaultTheme.shadows.large
        },
        assets: {
            logoUrl: apiTheme.logoUrl,
            headerImageUrl: apiTheme.headerImageUrl,
            footerLogoUrl: apiTheme.footerLogoUrl,
            faviconUrl: apiTheme.faviconUrl
        }
    };
};

export default defaultTheme;

File: backend/src/main/java/diploma/ecommerce/backend/shopbase/model/Theme.java
package diploma.ecommerce.backend.shopbase.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "themes")
public class Theme {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "theme_id")
    private Long id;

    // Базовые цвета
    @Column(name = "primary_color")
    private String primaryColor;

    @Column(name = "secondary_color")
    private String secondaryColor;

    @Column(name = "accent_color")
    private String accentColor;

    @Column(name = "text_color")
    private String textColor;

    @Column(name = "background_color")
    private String backgroundColor;

    // Шрифты
    @Column(name = "font_family")
    private String fontFamily;

    @Column(name = "heading_font_family")
    private String headingFontFamily;

    @Column(name = "body_font_family")
    private String bodyFontFamily;

    // Изображения
    @Column(name = "logo_url")
    private String logoUrl;

    @Column(name = "header_image_url")
    private String headerImageUrl;

    @Column(name = "footer_logo_url")
    private String footerLogoUrl;

    @Column(name = "favicon_url")
    private String faviconUrl;

    // Скругления элементов
    @Column(name = "button_radius")
    private String buttonRadius;

    @Column(name = "card_radius")
    private String cardRadius;

    @Column(name = "input_radius")
    private String inputRadius;

    // Цвета элементов
    @Column(name = "button_text_color")
    private String buttonTextColor;

    @Column(name = "footer_background_color")
    private String footerBackgroundColor;

    @Column(name = "footer_text_color")
    private String footerTextColor;

    // Цвета статусов
    @Column(name = "success_color")
    private String successColor;

    @Column(name = "error_color")
    private String errorColor;

    @Column(name = "warning_color")
    private String warningColor;

    @Column(name = "info_color")
    private String infoColor;

    // Интерактивные цвета
    @Column(name = "hover_color")
    private String hoverColor;

    @Column(name = "active_color")
    private String activeColor;

    // Тени
    @Column(name = "box_shadow")
    private String boxShadow;

    @Column(name = "card_shadow")
    private String cardShadow;

    // Временные метки
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    // Обратная связь с Tenant
    @OneToOne(mappedBy = "theme")
    private Tenant tenant;
}